services/src/test/java/edu/unc/lib/dl/cdr/services/rest/modify/SetAsPrimaryObjectIT.java

```diff
@@ -0,0 +1,186 @@
+
+import edu.unc.lib.dl.acl.exception.AccessRestrictionException;
+import edu.unc.lib.dl.acl.service.AccessControlService;
+import edu.unc.lib.dl.acl.util.AccessGroupSet;
+import edu.unc.lib.dl.acl.util.GroupsThreadStore;
+import edu.unc.lib.dl.fcrepo4.FileObject;
+import edu.unc.lib.dl.fcrepo4.FolderObject;
+import edu.unc.lib.dl.fcrepo4.PIDs;
+import edu.unc.lib.dl.fcrepo4.RepositoryObjectFactory;
+import edu.unc.lib.dl.fcrepo4.WorkObject;
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.rdf.PcdmModels;
+import edu.unc.lib.dl.sparql.JenaSparqlQueryServiceImpl;
+import edu.unc.lib.dl.test.TestHelper;
+
+/**
+ *
+ * @author harring
+ *
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration({"/spring-test/test-fedora-container.xml", "/spring-test/cdr-client-container.xml", "/set-as-primary-object-it-servlet.xml"})
+@WebAppConfiguration
+public class SetAsPrimaryObjectIT {
+
+    @Autowired
+    private WebApplicationContext context;
+    @Autowired
+    private RepositoryObjectFactory repositoryObjectFactory;
+    @Autowired
+    private AccessControlService aclService;
+    @Autowired
+    private JenaSparqlQueryServiceImpl queryService;
+
+    private MockMvc mvc;
+    private WorkObject parent;
+    private PID parentPid;
+    private FileObject fileObj;
+    private PID fileObjPid;
+
+    @Before
+    public void init() {
+
+        mvc = MockMvcBuilders
+                .webAppContextSetup(context)
+                .build();
+
+        TestHelper.setContentBase("http://localhost:48085/rest");
+
+        GroupsThreadStore.storeUsername("user");
+        GroupsThreadStore.storeGroups(new AccessGroupSet("adminGroup"));
+
+        fileObjPid = makePid();
+        fileObj = repositoryObjectFactory.createFileObject(fileObjPid, null);
+        parentPid = makePid();
+        parent = repositoryObjectFactory.createWorkObject(parentPid, null);
+    }
+
+    @After
+    public void tearDown() {
+        GroupsThreadStore.clearStore();
+    }
+
+    @Test
+    public void setPrimaryObjectTest() throws UnsupportedOperationException, Exception {
+        queryService.getModel().getResource(parentPid.getRepositoryPath())
+            .addProperty(PcdmModels.hasMember, fileObj.getResource());
+        parent.addMember(fileObj);
+
+        assertPrimaryObjectNotSet(parent);
+
+        MvcResult result = mvc.perform(put("/edit/setAsPrimaryObject/" + fileObjPid.getUUID()))
+                .andExpect(status().is2xxSuccessful())
+                .andReturn();
+
+        assertPrimaryObjectSet(parent, fileObj);
+
+        // Verify response from api
+        Map<String, Object> respMap = getMapFromResponse(result);
+        assertEquals(fileObjPid.getUUID(), respMap.get("pid"));
+        assertEquals("setAsPrimaryObject", respMap.get("action"));
+    }
+
+    @Test
+    public void testAuthorizationFailure() throws Exception {
+        doThrow(new AccessRestrictionException()).when(aclService)
```

Comment:
This test is good, but I'd suggest adding the fileObj as a member like in the previous test (maybe make a helper method?) so that it is a legitimate operation other than the access control exception